python/cuml/dask/ensemble/randomforestclassifier.py

```diff
@@ -0,0 +1,332 @@
+                
+        for i in range(remaining_est):
+            self.n_estimators_per_worker[i] = self.n_estimators_per_worker[i] + 1
+                    
+        ws = list(zip(workers, list(range(len(workers)))))
+                
+        self.rfs = {parse_host_port(worker):c.submit(RandomForestClassifier._func_build_rf, 
+                             n, self.n_estimators_per_worker[n], 
+                             max_depth, handle,
+                             max_features, n_bins,
+                             split_algo, split_criterion, 
+                             min_rows_per_node,
+                             bootstrap, bootstrap_features,
+                             type_model, verbose,
+                             rows_sample, max_leaves, quantile_per_tree,
+                             gdf_datatype, random.random(),
+                             workers=[worker])
+            for worker, n in ws}
+        
+        rfs_wait = list()
+        for r in self.rfs.values():
+            rfs_wait.append(r)
+                
+        wait(rfs_wait)
+        
+        
+    @staticmethod
+    def _func_build_rf(n, n_estimators, max_depth, handle,
+                             max_features, n_bins,
+                             split_algo, split_criterion, min_rows_per_node,
+                             bootstrap, bootstrap_features,
+                             type_model, verbose,
+                             rows_sample, max_leaves, quantile_per_tree,
+                             gdf_datatype, r):
+        
+        return cuRFC(n_estimators=n_estimators, max_depth=max_depth, handle=handle,
+                  max_features=max_features, n_bins=n_bins,
+                  split_algo=split_algo, split_criterion=split_criterion, min_rows_per_node=min_rows_per_node,
+                  bootstrap=bootstrap, bootstrap_features=bootstrap_features,
+                  type_model=type_model, verbose=verbose,
+                  rows_sample=rows_sample, max_leaves=max_leaves, quantile_per_tree=quantile_per_tree,
+                  gdf_datatype=gdf_datatype)
+                
+    
+    @staticmethod
+    def _fit(model, X_df, y_df, r): 
+        return model.fit(X_df, y_df)
+    
+    @staticmethod
+    def _predict(model, X, r): 
+        return model._predictGetAll(X)
+    
+    def fit(self, X, y):
+        """
+        Fit the input data to Random Forest classifier
+
+        Parameters
+        ----------
+        X : Acceptable format: dask-cudf. Dense matrix (floats or doubles) of shape (n_samples, n_features).
+        y : Acceptable format: dask-cudf. Dense matrix (floats or doubles) of shape (n_samples, 1)
+        """
+        c = default_client()
+
+        X_futures = c.sync(_extract_ddf_partitions, X)
+        y_futures = c.sync(_extract_ddf_partitions, y)
+                                       
+        f = list()
+        for w, xc in X_futures.items():     
```

Comment:
When you call `dask_df_utils`, you will just use `X_futures` and not `X_futures.items()`

Reply:
Per our discussion, leaving it as it is.